fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/StarRating.kt

```diff
@@ -0,0 +1,119 @@
+
+package org.mozilla.fenix.shopping.ui
+
+import androidx.annotation.VisibleForTesting
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.dp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.theme.FirefoxTheme
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.round
+import kotlin.math.roundToInt
+
+private const val NUM_STARS = 5
+private const val MAX_RATING = 5f
+private const val MIN_RATING = 0f
+
+/**
+ * UI for displaying star rating bar with maximum 5 stars.
+ */
+@Composable
+fun StarRating(
+    modifier: Modifier = Modifier,
+    value: Float,
+) {
+    val rating: Float = remember(value) {
+        max(min(MAX_RATING, value), MIN_RATING).roundToNearestHalf()
+    }
+    val contentDescription = contentDescription(rating = rating)
+
+    Row(
+        modifier = modifier.semantics {
+            this.contentDescription = contentDescription
+        },
+        horizontalArrangement = Arrangement.spacedBy(4.dp),
+    ) {
+        repeat(NUM_STARS) {
+            val star = if (it < rating && it + 1 > rating) {
+                R.drawable.ic_star_half
+            } else if (it < rating) {
+                R.drawable.ic_bookmark_filled
+            } else {
+                R.drawable.ic_bookmark_outline
+            }
+
+            Image(
+                painter = painterResource(id = star),
+                contentDescription = null,
+            )
+        }
+    }
+}
+
+@Composable
+private fun contentDescription(rating: Float): String {
+    val formattedRating: Number = remember(rating) { rating.removeDecimalZero() }
+
+    return stringResource(
+        R.string.review_quality_check_star_rating_content_description,
+        formattedRating,
+    )
+}
+
+/**
+ * Removes decimal zero if present and returns an Int, otherwise returns the same float as Number.
+ * e.g.4.0 becomes 4, 3.4 stays 3.4.
+ */
+@VisibleForTesting
+fun Float.removeDecimalZero(): Number {
+    val formattedRating: Number =
+        if (this % 1 == 0f) {
+            roundToInt()
+        } else {
+            this
+        }
+    return formattedRating
+}
+
+/**
+ * Rounds the float to the nearest half instead of the whole number.
+ * e.g 4.6 becomes 4.5 instead of 5.
+ */
+@VisibleForTesting
+fun Float.roundToNearestHalf(): Float =
+    round(this * 2) / 2f
```

Comment:
Let's move these to their own extension file (Float.kt?) similar to how we have various `String.kt` extension files